### ISSUE TITLE
mgcb-editor Font Family specified is not supported by this system

### ISSUE THREAD

**Issue by gavrilyuc** [MonoGame/MonoGame#7617]
I have the problem. MacOS 11.6. I used this [Guide](https://docs.monogame.net/articles/getting_started/1_setting_up_your_development_environment_macos.html)

Installed Visual Studio Mac and mrcb-editor (before to github issue I was installed mgcb by `dotnet tool install -g dotnet-mgcb`) and created new project.

After that I tried to open `Content.mgcb` file but I get:
![image](https://user-images.githubusercontent.com/9469728/142619103-23979087-60a2-4fc1-a460-4c1d4ae0ffe5.png)

And I have some IDE errors:

First error from list:
`Could not open file "/Users/gavrilyuc/Desktop/Pandora.Puzzle/Pandora.Puzzle.IPhone/Content/Content.mgcb"`
```sh
Не удалось открыть файл "/Users/gavrilyuc/Desktop/Pandora.Puzzle/Pandora.Puzzle.IPhone/Content/Content.mgcb".
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ArgumentOutOfRangeException: Font Family specified is not supported by this system
Parameter name: familyName
Actual value was .AppleSystemUIFont.
  at Eto.GtkSharp.Drawing.FontFamilyHandler.Create (System.String familyName) [0x001a3] in <24aa521f412944ac8c81a57908c83325>:0 
  at Eto.Drawing.FontFamily..ctor (System.String familyName) [0x0001f] in <2a34c79ad30740c2858e3ee2ae987ec5>:0 
  at Eto.Drawing.Font..ctor (System.String family, System.Single size, Eto.Drawing.FontStyle style, Eto.Drawing.FontDecoration decoration) [0x0000c] in <2a34c79ad30740c2858e3ee2ae987ec5>:0 
  at MonoGame.IDE.VisualStudioForMac.MGPad..ctor () [0x00028] in <da0baa3a6e9b4efab45510dde1193402>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeConstructorInfo.InternalInvoke(System.Reflection.RuntimeConstructorInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x00005] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:936 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x00015] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:944 
  at System.RuntimeType.CreateInstanceMono (System.Boolean nonPublic, System.Boolean wrapExceptions) [0x00095] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/ReferenceSources/RuntimeType.cs:185 
  at System.RuntimeType.CreateInstanceSlow (System.Boolean publicOnly, System.Boolean wrapExceptions, System.Boolean skipCheckThis, System.Boolean fillCache) [0x00009] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/ReferenceSources/RuntimeType.cs:155 
  at System.RuntimeType.CreateInstanceDefaultCtor (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Boolean wrapExceptions, System.Threading.StackCrawlMark& stackMark) [0x00027] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/rttype.cs:5770 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic, System.Boolean wrapExceptions) [0x00039] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/activator.cs:206 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/activator.cs:190 
  at Mono.Addins.RuntimeAddin.CreateInstance (System.String typeName, System.Boolean throwIfNotFound) [0x00014] in /Users/builder/azdo/_work/2/s/main/external/mono-addins/Mono.Addins/Mono.Addins/RuntimeAddin.cs:453 
  at MonoDevelop.Ide.Codons.PadCodon.CreatePad () [0x0000b] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Codons/PadCodon.cs:142 
  at MonoDevelop.Ide.Codons.PadCodon.InitializePadContent (MonoDevelop.Ide.Gui.IPadWindow window) [0x00008] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Codons/PadCodon.cs:85 
  at MonoDevelop.Ide.Gui.PadWindow.CreateContent () [0x00008] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/IPadContainer.cs:305 
  at MonoDevelop.Ide.Gui.PadWindow.Activate (System.Boolean giveFocus) [0x00000] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/IPadContainer.cs:297 
  at MonoDevelop.Ide.Gui.Pad.BringToFront (System.Boolean grabFocus) [0x00011] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Pad.cs:128 
  at MonoDevelop.Ide.Gui.Pad.BringToFront () [0x00000] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Pad.cs:122 
  at MonoGame.IDE.VisualStudioForMac.MGCBPadEditor.Launch (System.String[] files) [0x00036] in <da0baa3a6e9b4efab45510dde1193402>:0 
  at MonoDevelop.Ide.Gui.Documents.DocumentManager.RealOpenFile (MonoDevelop.Core.ProgressMonitor monitor, MonoDevelop.Ide.Gui.FileOpenInformation openFileInfo, MonoDevelop.Core.Instrumentation.ITimeTracker`1[T] timer) [0x007b5] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Documents/DocumentManager.cs:557 
```

Second: `An unhandled exception has occurred. Terminating Visual Studio? False`
```sh
An unhandled exception has occurred. Terminating Visual Studio? False
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ArgumentOutOfRangeException: Font Family specified is not supported by this system
Parameter name: familyName
Actual value was .AppleSystemUIFont.
  at Eto.GtkSharp.Drawing.FontFamilyHandler.Create (System.String familyName) [0x001a3] in <24aa521f412944ac8c81a57908c83325>:0 
  at Eto.Drawing.FontFamily..ctor (System.String familyName) [0x0001f] in <2a34c79ad30740c2858e3ee2ae987ec5>:0 
  at Eto.Drawing.Font..ctor (System.String family, System.Single size, Eto.Drawing.FontStyle style, Eto.Drawing.FontDecoration decoration) [0x0000c] in <2a34c79ad30740c2858e3ee2ae987ec5>:0 
  at MonoGame.IDE.VisualStudioForMac.MGPad..ctor () [0x00028] in <da0baa3a6e9b4efab45510dde1193402>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeConstructorInfo.InternalInvoke(System.Reflection.RuntimeConstructorInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x00005] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:936 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x00015] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:944 
  at System.RuntimeType.CreateInstanceMono (System.Boolean nonPublic, System.Boolean wrapExceptions) [0x00095] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/ReferenceSources/RuntimeType.cs:185 
  at System.RuntimeType.CreateInstanceSlow (System.Boolean publicOnly, System.Boolean wrapExceptions, System.Boolean skipCheckThis, System.Boolean fillCache) [0x00009] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/ReferenceSources/RuntimeType.cs:155 
  at System.RuntimeType.CreateInstanceDefaultCtor (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Boolean wrapExceptions, System.Threading.StackCrawlMark& stackMark) [0x00027] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/rttype.cs:5770 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic, System.Boolean wrapExceptions) [0x00039] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/activator.cs:206 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/activator.cs:190 
  at Mono.Addins.RuntimeAddin.CreateInstance (System.String typeName, System.Boolean throwIfNotFound) [0x00014] in /Users/builder/azdo/_work/2/s/main/external/mono-addins/Mono.Addins/Mono.Addins/RuntimeAddin.cs:453 
  at MonoDevelop.Ide.Codons.PadCodon.CreatePad () [0x0000b] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Codons/PadCodon.cs:142 
  at MonoDevelop.Ide.Codons.PadCodon.InitializePadContent (MonoDevelop.Ide.Gui.IPadWindow window) [0x00008] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Codons/PadCodon.cs:85 
  at MonoDevelop.Ide.Gui.DefaultWorkbench.CreatePadContent (System.Boolean force, MonoDevelop.Ide.Codons.PadCodon padCodon, MonoDevelop.Ide.Gui.PadWindow window, MonoDevelop.Components.Docking.DockItem item) [0x0000c] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Shell/DefaultWorkbench.cs:1523 
  at MonoDevelop.Ide.Gui.DefaultWorkbench.<AddPad>g__OnItemContentRequired|150_0 (System.Object sender, System.EventArgs args) [0x0003d] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Shell/DefaultWorkbench.cs:1598 
  at MonoDevelop.Components.Docking.DockItem.SetupContent (System.Object ob, System.EventArgs args) [0x00026] in /Users/builder/azdo/_work/2/s/main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Docking/DockItem.cs:223 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:395 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0007e] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:409 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Reflection/MethodBase.cs:53 
  at System.Delegate.DynamicInvokeImpl (System.Object[] args) [0x000e1] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System/Delegate.cs:465 
  at System.MulticastDelegate.DynamicInvokeImpl (System.Object[] args) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System/MulticastDelegate.cs:67 
  at System.Delegate.DynamicInvoke (System.Object[] args) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/mono-x64/mcs/class/corlib/System/Delegate.cs:410 
  at GLib.Signal.ClosureInvokedCB (System.Object o, GLib.ClosureInvokedArgs args) [0x0005c] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/gtk-sharp-None/glib/Signal.cs:200 
  at GLib.Signal+SignalClosure.Invoke (GLib.ClosureInvokedArgs args) [0x00009] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/gtk-sharp-None/glib/SignalClosure.cs:114 
  at GLib.Signal+SignalClosure.MarshalCallback (System.IntPtr raw_closure, GLib.Value* return_val, System.UInt32 n_param_vals, GLib.Value* param_values, System.IntPtr invocation_hint, System.IntPtr marshal_data) [0x0003e] in /Users/builder/jenkins/workspace/build-package-osx-mono/2020-02/external/bockbuild/builds/gtk-sharp-None/glib/SignalClosure.cs:132 
 ```
 
Third:
 ```sh
 Unable to find execution handler. command.Type=MonoDevelop.IPhone.IPhoneExecutionCommand executionHandlers.Count=9 executionHandlers=Mono, Native, DotNetCore, MonoMac, AspNetCore, AspNet, IPhone, MonoDroid, AzureFunctions
 ```
 

When I trying open `mgcb-editor` by terminal I saw opening shortcut icon then the icon was hided.

How can I using Visual Studio Mac with mgcb editor?

Additional about the trouble:
I installed (again) [sf fonts](https://developer.apple.com/fonts) but it does not help me.
`mgcb` was installed but it does not help me by the [Guide](https://docs.monogame.net/articles/tools/mgcb.html).

**Comment by gavrilyuc** [comment]
![image](https://user-images.githubusercontent.com/9469728/142720813-78b6e470-d0bd-4357-94be-abde5e7956a2.png)

**Comment by gavrilyuc** [comment]
I can't build this because your projects is .net472 and I can't using windows environment. How can help to fix this one?

**Comment by stromkos** [comment]
You are attempting to run an IPhone targeted application directly on a Mac. Not supported.

Create your project using the `DesktopGL` profile to run and debug on your mac computer. 

Only then do you create the IOS project. Merging the `Game1.cs` and copying the rest of the .cs file to the new project.

Ви намагаєтеся запустити цільову програму для iPhone безпосередньо на Mac. Не підтримується.

Створіть свій проект за допомогою профілю DesktopGL для запуску та налагодження на комп’ютері Mac.

Тільки після цього ви створюєте проект IOS. Об’єднання `Game1.cs` та копіювання решти файлу .cs до нового проекту.
Vy namahayetesya zapustyty tsilʹovu prohramu dlya iPhone bezposerednʹo na Mac. Ne pidtrymuyetʹsya.

Stvoritʹ sviy proekt za dopomohoyu profilyu DesktopGL dlya zapusku ta nalahodzhennya na kompʺyuteri Mac.

Tilʹky pislya tsʹoho vy stvoryuyete proekt IOS. Obʺyednannya `Game1.cs` ta kopiyuvannya reshty faylu .cs do novoho proektu.

**Comment by harry-cpp** [comment]
@stromkos You misunderstood him, mgcb-editor (and mgcb pad, which was mgcb-editor inside vs for mac) are crashing for him with different exceptions 😅, and he couldn't open them to fix the .spritefont file which was pointing to probably some windows only font.

The issues with mgcb-editor are fixed in 3.8.1
